pgjdbc/src/main/java/org/postgresql/core/PGStream.java

```diff
@@ -355,11 +344,10 @@ public void sendChar(int val) throws IOException {
    * @throws IOException if an I/O error occurs
    */
   public void sendInteger4(int val) throws IOException {
-    int4Buf[0] = (byte) (val >>> 24);
-    int4Buf[1] = (byte) (val >>> 16);
-    int4Buf[2] = (byte) (val >>> 8);
-    int4Buf[3] = (byte) (val);
-    pgOutput.write(int4Buf);
```

Comment:
Is pgOutput always buffered internally?

Reply:
based on my understanding, its always buffered except when its a char (single byte). Good observation as this can increase IO compromising performance? Is that what you were getting to? 

Can create a local buffer within this scope and write once. Not sure if this is what you were targeting?